internal/controller/bucket_controller_test.go

```diff
@@ -608,6 +608,43 @@ func TestBucketReconciler_reconcileSource_generic(t *testing.T) {
 				*conditions.UnknownCondition(meta.ReadyCondition, "foo", "bar"),
 			},
 		},
+		{
+			name:       "Observes incompatible STS provider",
+			bucketName: "dummy",
+			beforeFunc: func(obj *bucketv1.Bucket) {
+				obj.Spec.Provider = "some-provider"
```

Comment:
> But we can explicitly set generic and have that in the error message.

This solution won't work when introducing the `ldap` STS provider for the `generic` bucket provider

> Maybe add a TODO to change the provider to generic when it becomes a valid value?

:+1: 

By the way, I already have the `ldap` provider ready, just didn't open a PR yet because I want to stop pushing branches to `matheuscscp/source-controller` and start using the maintainer benefit of pushing to `fluxcd/source-controller` :sweat_smile:

Reply:
> This solution won't work when introducing the ldap STS provider for the generic bucket provider

Nevermind, I just need to use the `aws` STS provider, and then change the error message when the `generic` bucket provider starts accepting an STS provider (`ldap`)

Reply:
Fixed, thanks!